requests_toolbelt/utils/dump.py

```diff
@@ -3,19 +3,116 @@
 
 from requests import compat
 
-
-__all__ = ('dump_response', 'dump_all')
+__all__ = ('dump_response', 'dump_all', 'Sanitizer', 'HeaderSanitizer')
 
 HTTP_VERSIONS = {
     9: b'0.9',
     10: b'1.0',
     11: b'1.1',
 }
 
+#: List of sensitive headers copied from
+#: https://github.com/google/har-sanitizer
+SENSITIVE_HEADERS = {
+    "state",
+    "shdf",
+    "usg",
+    "password",
+    "email",
+    "code",
+    "code_verifier",
+    "client_secret",
+    "client_id",
+    "token",
+    "access_token",
+    "authenticity_token",
+    "id_token",
+    "appid",
+    "challenge",
+    "facetid",
+    "assertion",
+    "fcparams",
+    "serverdata",
+    "authorization",
+    "auth",
+    "x-client-data",
+    "samlrequest",
+    "samlresponse"
+}
+
 _PrefixSettings = collections.namedtuple('PrefixSettings',
                                          ['request', 'response'])
 
 
+class Sanitizer(object):
+    """Performs no sanitation"""
+
+    def request_header(self, name, value):
```

Comment:
I believe that we should sanitize `headers` not individual pairs. How would someone, for example, strip a header name-value pair if they wanted to?

Reply:
I did it like this because iterating over the headers and values is non-trivial, and I thought it would be more useful if you didn't have to figure out how to do that every time you want to implement a sanitizer. Also, since headers is not a dict, what should this method return in order to replace it? Or should it modify the headers object in-place (is that possible)?

With regards to stripping a name-value pair completely, I figured that was a very uncommon use-case that might not be worth supporting.

I can change it to work on the full headers, but then I need help understanding the questions above.

Reply:
> Also, since headers is not a dict, what should this method return in order to replace it? 

So headers is going to be a requests-specific dictionary implementation: `CaseInsensitiveDict`. We can, do this:

```
sanitized_headers = headers.copy()
for name in headers:
    if self.should_sanitize_header(name):
        sanitized_headers[name] = REDACTED_VALUE
    elif self.should_strip_header(name):
        del sanitized_headers[name]
```

This would then allow users to define `should_sanitize_header` and `should_strip_header` which only needs to operate on the header name and return True/False. This could be the default implementation for the `_header` sanitization functions since calling the `should_*_header` would raise exceptions if not implemented by default.

Reply:
Ok, I think I get the gist of your suggestion. I will try to make some changes to move in this direction, and we can see where that gets us.